lib/memo_wise.rb

```diff
@@ -3,6 +3,46 @@
 require "memo_wise/version"
 
 module MemoWise
-  class Error < StandardError; end
-  # Your code goes here...
+  def initialize(*values)
+    @_memo_wise = {}
+    super
+  end
+
+  def self.prepended(target)
+    class << target
+      # Implements memoization for the given method name.
+      #
+      # @param method_name [Symbol]
+      #   Name of method for which to implement memoization.
+      def memo_wise(method_name)
```

Comment:
How would you feel about raising `ArgumentError` if `method_name` is NOT a `Symbol`, and adding specs? I see benefits in:
1. I can suggest some optimizations knowing for sure we have a Symbol
2. We have a documented and tested API error case surface, to add and update over time